src/test/java/com/endava/internship/collections/StudentListTest.java

```diff
@@ -0,0 +1,551 @@
+package com.endava.internship.collections;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+
+import java.util.ListIterator;
+import java.util.NoSuchElementException;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class StudentListTest {
+    List<Student> studentList;
+
+    @BeforeEach
+    void setUp() {
+        studentList = new StudentList<>();
+        Student st1 = new Student("Unu", LocalDate.of(2001, 1, 1), "Student ONE");
+        Student st2 = new Student("Doi", LocalDate.of(2004, 1, 1), "Student TWO");
+        Student st3 = new Student("Trei", LocalDate.of(2006, 1, 1), "Student TREE");
+        studentList.add(st1);
+        studentList.add(st2);
+        studentList.add(st3);
+    }
+
+    @Test
+    void shouldThrowIllegalArgumentExceptionWithInvalidParameter() {
+        Throwable exceptionThatWasThrown = assertThrows(IllegalArgumentException.class, () -> new StudentList<Student>(-1));
+        assertEquals(exceptionThatWasThrown.getMessage(), "Illegal Capacity: " + "-1");
+    }
+
+    @Test
+    void shouldCreateAListWithDefaultCapacity() {
```

Comment:
you cannot prove that the list has been create with default capacity. 
Better remove this test

Reply:
Okay I can delete all this test but the coverage won't 100%

Reply:
Done.